src/helpers.rs

```diff
@@ -81,3 +82,25 @@ pub(crate) fn resolve_pallet_path(path: Option<String>) -> PathBuf {
 		}
 	}
 }
+
+pub(crate) fn is_git_repo_with_commits(repo_path: &Path) -> anyhow::Result<bool> {
+	match Repository::open(repo_path) {
+		Ok(repo) => {
+			let mut status_opts = git2::StatusOptions::new();
+			status_opts.include_untracked(false);
+			let statuses = repo.statuses(Some(&mut status_opts))?;
+
+			// Check if there are no changes to commit
+			Ok(if !statuses.iter().any(|s| s.status() != git2::Status::CURRENT) {
+				true
+			} else {
+				log::warning(format!("Repository has uncommitted changes."));
+				false
+			})
+		},
+		Err(e) => {
+			log::error(format!("Failed to open repository: {}", e));
```

Comment:
We can implement a try strategy here.. the path that you write probably contains a `lib.rs` and it's probably the one the user intended instead of spelling out the entire path upto `lib.rs`. In which case, we can try to look for a `lib.rs` if the provided path is a folder, one of either `[$runtime, $runtime/src]` and try calling execute -> which will in-turn check automatically if the contained `lib.rs` is a proper runtime file. 
in this case, i think we should also print at the end of the addition which file was edited in order to fully dummy-proof it. This will make a good issue as well.